test/crit/main.go

```diff
@@ -0,0 +1,89 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/checkpoint-restore/go-criu/v5/crit"
+)
+
+func main() {
+	imgs := getImgs()
+	for _, img := range imgs {
+		log.Println("=== ", img)
+		testImg := fmt.Sprintf("%s.test.img", filepath.Base(img))
+		c := crit.New(img, testImg, "", false, false)
+		// Decode the binary image file
+		decodedImg, err := c.Decode()
+		if err != nil {
+			log.Fatal("[DECODE]: ", err)
+		}
+		// Encode it into test binary image file
+		err = c.Encode(decodedImg)
+		if err != nil {
+			log.Fatal("[ENCODE]: ", err)
+		}
+		// Open and compare original and test files
+		imgBytes, err := os.ReadFile(img)
+		if err != nil {
+			log.Fatal(err)
+		}
+		testImgBytes, err := os.ReadFile(testImg)
+		if err != nil {
+			log.Fatal(err)
+		}
+		if !bytes.Equal(imgBytes, testImgBytes) {
+			log.Fatal("[RECODE]: files do not match")
+		}
+		log.Println("=== PASS")
+	}
+}
+
+func getImgs() []string {
+	/* Certain image files generated by CRIU do not
+	use the protobuf format and contain raw binary
+	data. Some image files are also generated using
+	external tools (ifaddr, route, tmpfs). As these
+	images cannot be processed by CRIT, they are
+	excluded from the tests. */
```

Comment:
The convention for [variable comments](https://go.dev/doc/comment#var) is using `//`.

Reply:
I initially thought this would be more of a developer-specific comment, hence the multi line syntax